deploy/config-spec/properties.yaml

```diff
@@ -0,0 +1,144 @@
+version: 0.1.0
+spec:
+  units:
+    - unit: &unitPort
+        name: "port"
+        regex: "^([0-9]{1,4}|[1-5][0-9]{4}|6[0-4][0-9]{3}|65[0-4][0-9]{2}|655[0-2][0-9]|6553[0-5])$"
+
+    - unit: &unitUrl
+        name: "url"
+        regex: "^((https?|ftp|file)://)?[-a-zA-Z0-9+&@#}/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]"
+        examples:
+          - "https://www.stackable.de/blog/"
+
+    - unit: &unitCapacity
+        name: "capacity"
+        regex: "^[1-9]\\d*$"
+
+    - unit: &unitMilliseconds
+        name: "milliseconds"
+        regex: "^[1-9]\\d*$"
+
+properties:
+  - property: &zookeeperConnect
+      propertyNames:
+        - name: "zookeeper.connect"
+          kind:
+            type: "file"
+            file: "server.properties"
+      datatype:
+        type: "string"
+        unit: *unitUrl
+      defaultValues:
+        - fromVersion: "0.0.0"
+          value: "localhost:2181"
+      roles:
+        - name: "broker"
+          required: true
+      asOfVersion: "0.0.0"
+      description: "The zookeeper connection string"
+
+  - property: &zookeeperTimeout
+      propertyNames:
+        - name: "zookeeper.connection.timeout.ms"
+          kind:
+            type: "file"
+            file: "server.properties"
+      datatype:
+        type: "integer"
+        unit: *unitMilliseconds
+      defaultValues:
+        - fromVersion: "0.0.0"
+          value: "18000"
+      roles:
+        - name: "broker"
+          required: true
+      asOfVersion: "0.0.0"
+      description: "Zookeeper connection timeout in milliseconds."
+
+  - property: &opaAuthorizerClassName
```

Comment:
This can in theory be any authorizer, it is not an opa specific setting.

Reply:
Yeah not sure how to handle this right now. The class name (as well as according properties) may differ. How do we want to separate that?

Reply:
I think we'll need to work on that as part of the overall security workstream. We'll need to offer both options:

a) Referencing an Opa Authorizer and then our operator will take care of the rest
b) Directly setting a custom authorizer if the customer requires one (probably via safety valves)